src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -225,6 +225,46 @@ def __init__(self, value: Any, datatype: Optional[DataType] = None) -> None:
             self.datatype = infer_type(value)
 
 
+class Interval(Expression):
+    def __init__(
+        self,
+        year: Optional[int] = 0,
+        quarter: Optional[int] = 0,
+        month: Optional[int] = 0,
+        week: Optional[int] = 0,
+        day: Optional[int] = 0,
+        hour: Optional[int] = 0,
+        minute: Optional[int] = 0,
+        second: Optional[int] = 0,
+        millisecond: Optional[int] = 0,
+        microsecond: Optional[int] = 0,
+        nanosecond: Optional[int] = 0,
+    ) -> None:
+        super().__init__()
+        self.year = year
+        self.quarter = quarter
+        self.month = month
+        self.week = week
+        self.day = day
+        self.hour = hour
+        self.minute = minute
+        self.second = second
+        self.millisecond = millisecond
+        self.microsecond = microsecond
+        self.nanosecond = nanosecond
+
+    def sql(self) -> str:
+        return (
+            f"INTERVAL '{self.year} year, {self.quarter} quarter, {self.month} month, "
+            f"{self.week} week, {self.day} day, {self.hour} hour, {self.minute} minute, "
+            f"{self.second} second, {self.millisecond} millisecond, {self.microsecond} microsecond, "
+            f"{self.nanosecond} nanosecond'"
```

Comment:
If a value is 0 can it be safely left out?

Reply:
We could detect which values are non-zero and only include those, but that parsing isn't really necessary given adding the "0" values doesn't cost us anything and is lesser lines of code.